NitroxClient/GameLogic/PlayerManager.cs

```diff
@@ -0,0 +1,46 @@
+﻿using System;
+using System.Collections.Generic;
+
+namespace NitroxClient.GameLogic
+{
+    public class PlayerManager
+    {
+        private readonly Dictionary<string, RemotePlayer> playersById = new Dictionary<string, RemotePlayer>();
+
+        public RemotePlayer this[string playerId]
+        {
+            get
+            {
+                return FindPlayer(playerId);
+            }
+            set
+            {
+                if (value == null)
+                    RemovePlayer(playerId);
+                else
+                    playersById[playerId] = value;
+            }
+        }
+
+        public RemotePlayer FindPlayer(string playerId, bool createPlayer = false)
+        {
+            RemotePlayer player;
+            if (!playersById.TryGetValue(playerId, out player) && createPlayer)
+                player = playersById[playerId] = new RemotePlayer(playerId);
+            return player;
+        }
+
+        public void ForPlayer(string playerId, Action<RemotePlayer> action, bool createPlayer = false)
+        {
+            var player = FindPlayer(playerId, createPlayer);
+            if (player != null)
```

Comment:
Please always use curly braces - no one liners like this.  There are a few other places in the PR - do a quick glance through.